features/mesh/services/Index.feature

```diff
@@ -17,8 +17,8 @@ Feature: mesh / services / index
       """
       body:
         items:
-        - name: fake-service-1
-        - name: fake-service-2
+        - name: service-1_external
+        - name: service-2_internal
```

Comment:
The current state of mocks regarding the particular interaction and relation between `ServiceInsight`s and `ExternalService`s made it really hard to build out the UI properly and write reasonable tests. It was far too easy to look at _very_ inconsistent views in terms of their data.

Reply:
Yeah I just think its a little strange that there are some PRs where you are super insistent on mocking out all the data even though you don't need to, and in this case it looks like it would be good to explicitly mock out more data (not all of it) and you've chosen not to. Not a biggie/blocker.

Reply:
How exactly are those things related?

Sure, it’s possible to mock out the data explicitly here and not rely on the suffix mechanism for the service type, and maybe that’s the better approach for tests, but my point about using explicit mocks is not an absolute criteria. It’s for cases where having mocks merge into default data is making the act of writing the test unnecessarily more complex and complicated. This is not such a case as there are little to no array fields on the data under test here. That’s a different matter for `DataplaneOverview`s for example that have lots of array data and specifically of the kind where order matters.

Reply:
>How exactly are those things related?

Because they are both to do with the amount of data we mock out, one approach (this one) as far as I understand it is an approach to mock out less when we probably should mock what we test for, and another approach (in other PRs) is mocking out more than necessary i.e. we mock out what we don't test for.

Again this isn't a blocker, its just something I noticed. That being said I would rather we mocked out the things we test for/assert against, and not mock out the things that we don't test for/assert against.

Reply:
To be clear: I didn’t add the suffixes to this test in order to avoid mocking out stuff. The way the tests were written makes them pass with or without me doing so (in fact I removed the suffixes now).

The point of supporting these suffixes is to get more sensible data in general.

Reply:
Ah ok I thought you'd added them to get specific data.

> The point of supporting these suffixes is to get more sensible data in general.

Yeah its great for building stuff out where we can't really mock data easily without changing code. Being able to add a `_internal` to the URL is really handy but we shouldn't use it in place of mocking out the data we test for, and from what I've understood from this conversation, we aren't, so we are all good.